CLDR-14607 move metazoneAliases under supplemental

srl295 opened the pull request:
CLDR-14607

- [ ] This PR completes the ticket.

<!--
Thank you for your pull request.
Please see http://cldr.unicode.org/index/process for general
information on contributing to CLDR.

1. Make sure the ticket is filed at
https://unicode-org.atlassian.net/projects/CLDR/
2. Update the PR title and first line of this
message to include the ticket ID (CLDR-_____)
3. You will be automatically asked to sign the contributors’
license before the PR is accepted.
- sign: https://cla-assistant.io/unicode-org/cldr
- license: http://www.unicode.org/copyright.html#License
-->

srl295 commented:
WIP: needs dtd fix

macchiati commented:
Just sent out a message about this to core

Mark


On Fri, Sep 3, 2021 at 9:51 AM Steven R. Loomis ***@***.***>
wrote:

> WIP: needs dtd fix
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/unicode-org/cldr/pull/1485#issuecomment-912675116>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACJLEMEGYLMJQSAEL7LRR53UAD4IXANCNFSM5DMDU64Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

srl295 commented:
yeah, crossed in the mail… 



> El sep. 3, 2021, a las 11:52 a. m., Mark Davis ***@***.***> escribió:
> 
> 
> Just sent out a message about this to core
> 
> Mark
> 
> 
> On Fri, Sep 3, 2021 at 9:51 AM Steven R. Loomis ***@***.***>
> wrote:
> 
> > WIP: needs dtd fix
> >
> > —
> > You are receiving this because your review was requested.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/unicode-org/cldr/pull/1485#issuecomment-912675116>,
> > or unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/ACJLEMEGYLMJQSAEL7LRR53UAD4IXANCNFSM5DMDU64Q>
> > .
> > Triage notifications on the go with GitHub Mobile for iOS
> > <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> > or Android
> > <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
> >
> >
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub <https://github.com/unicode-org/cldr/pull/1485#issuecomment-912675841>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAGQZMZ6KGPAKGDDKTVYVF3UAD4OBANCNFSM5DMDU64Q>.
>

srl295 commented on tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/PathHeader.txt:
```diff
@@ -417,6 +417,7 @@
 //supplementalData/metaZones/metazoneInfo/timezone[@type="%A"]/usesMetazone/_mzone			; Supplemental ; Metazone ; $1 ; -∞..∞ ; HIDE
 
 //supplementalData/metaZones/mapTimezones[@type="%A"]/mapZone[@other="%A"][@territory="%A"]/_type			; Supplemental ; Metazone ; $1 ; $2-$3 ; HIDE
+//supplementalData/metaZones/shortIds/shortId[@name="%A"]_alias    ; Supplemental ; Metazone ; $1 ; $2-$3 ; HIDE
```
@macchiati What is the test expecting here?

>     TestCompletenessNonLdmlDtd {
>      Error: (TestPathHeader.java:1318) //supplementalData/metaZones/shortIds/shortId[@name="%A"]/_deprecated: Unknown path header[], Supplemental      Unknown metaZones       shortIds/shortId[@name="acre"]/_deprecated
>        No match in PathHeader.txt for //supplementalData/metaZones/shortIds/shortId[@name="acre"]/_deprecated
>        You get only one message for all paths matching //supplementalData/metaZones/shortIds/shortId[@name="%A"]/_deprecated
>        For example, check to see if the field in PathHeader.txt is in PathHeader.PageId.
>        If not, either correct PathHeader.txt or add it to PageId
>        If you have a value attribute, you will need extra _ characters. The value attribute will show at the end with prefixed _, eg [...]/_skeleton.If there can be a value for the path then that element will add _.

macchiati replied:
Every kind of path needs a match in Pathheader.txt. Easiest way to do this is to look for a path like the one you have.

EG //supplementalData/metaZones/mapTimezones[@type="%A"]/mapZone[@other="%A"][@territory="%A"]/_type			; Supplemental ; Metazone ; $1 ; $2-$3 ; HIDE

WARNING: the longest matches have to come first in the file, and you need to have the various combinations of #IMPLIED attributes that appear. That's why you see:

//supplementalData/metaZones/metazoneInfo/timezone[@type="%A"]/usesMetazone[@from="%A"][@to="%A"]/_mzone			; Supplemental ; Metazone ; $1 ; $2-$3 ; HIDE
//supplementalData/metaZones/metazoneInfo/timezone[@type="%A"]/usesMetazone[@from="%A"]/_mzone			; Supplemental ; Metazone ; $1 ; $2..∞ ; HIDE
//supplementalData/metaZones/metazoneInfo/timezone[@type="%A"]/usesMetazone[@to="%A"]/_mzone			; Supplemental ; Metazone ; $1 ; -∞..$2 ; HIDE
//supplementalData/metaZones/metazoneInfo/timezone[@type="%A"]/usesMetazone/_mzone			; Supplemental ; Metazone ; $1 ; -∞..∞ ; HIDE

For the attributes for shortId, only preferred may or might not occur. So two patterns should suffice.

macchiati replied:
You have $2-$3, bu there is only one variable in the regex (%A), so only a $1. 

Best to look at the bcp47 equivalents for the pattern.

macchiati replied:
The test results show what the paths turn into. The value parameters turn into separate items
//supplementalData/metaZones/metazoneIds/metazoneId[@shortId="%A"]/_deprecated
//supplementalData/metaZones/metazoneIds/metazoneId[@shortId="%A"]/_longId

macchiati replied:
So you need to provide values for these. The values for all three can be done with the following, I think.

//supplementalData/metaZones/shortIds/shortId[@name="%A"]_%E    ; Supplemental ; Metazone ; $1 ; $2 ; HIDE

srl295 commented:
I think it's ready to go , modulo some test fixes

srl295 commented:
@yumaoka comments:
> - "description" attribute does not make sense here. All entries in the proposed data use "Metazone". But this is under <metaZones><metazoneIds>
> - We may define since="40" as default value.

yumaoka commented on common/supplemental/metaZones.xml:
```diff
@@ -1892,6 +1892,167 @@ For terms of use, see http://www.unicode.org/copyright.html
 		</mapTimezones>
+
+		<shortIds>
+            <shortId name="acre" description="Metazone" alias="Acre" since="40"/>
```
description="Metazone" is redundant. This data is now under <metaZones><shortId>, so it's metazone ID obviously. We could add meaningful description later, for example "Time used by Acre state in Brazil".

yumaoka replied:
Because inception of this data is CLDR 40, we could default to since="40" and use since="xx" only for future additions.

srl295 replied:
The display name of the metazone should come from CLDR.

Do we want this supplemental item to be "meritable facts about this metazone"? or is it strictly "short aliases"? If it's the former, we should have a different structure and the element should be `<metazoneInfo>` or such. 

We don't currently have any chunk of data that exists one time for each metazone.

yumaoka replied:
I actually prefer to have a complete list of metazones - with long/short IDs.
At this moment, if you want to get all valid metazone IDs, you have to walk through the mapping data and collect all of unique IDs.

srl295 replied:
@yumaoka we can do that, but that's a different type of structure. right now `metazoneInfo` AND `mapTimezones` each have mappings to metazones. I'd assume we don't want to actually change those existing items?

If we want an element that is the canonical list of metazones, I'd name it so:


```xml
<metazoneIds>
  <metazoneId type="Europe_Western" alias="euwe" since="41"/> <!-- assuming a non-default 'since' -->
</metazoneIds>
```

This structure and naming would allow for other metadata on a per-metazone basis.

yumaoka replied:
Maybe something like below?

```
<metaZones>
  <metazoneList>
    <metazone type="Acre" short="acre" description="Time for Acre, Brazil"/>
    ....
  </metazoneList>
  <metazoneInfo>
    <!-- tz/mz mapping data -->
    ...
  </metazoneInfo>
</metaZoens>
```

If we did not have <metaZones> element, we would want to use `<metazones>` instead of `<metazoneList>` above. The element <metazone> describes the definition of each metazone - comes with long id and short id.

macchiati commented:
That is a reasonable structure (wish this discussion had happened some time
ago though). It would correspond to the following.

(minor changes: we use the plural instead of list usually, and then the
singular, so metazoneIds/metazoneId. And it is best to avoid the generic
'type' for clarity)

<!ELEMENT metaZones ( metazoneIds?, mapTimezones?, shortIds? ) >

<!ELEMENT *metazoneIds* ( *metazoneId** ) >

<!ELEMENT metazoneId EMPTY >
<!ATTLIST metazoneId *shortId* NMTOKEN #REQUIRED >
    ***@***.***:regex/[a-z][a-z][a-z][a-z]-->
<!ATTLIST metazoneId *longId* CDATA #IMPLIED >
    ***@***.***:any-->
    ***@***.***>
<!ATTLIST metazoneId description CDATA #REQUIRED >
    ***@***.***:any-->
    ***@***.***>
<!ATTLIST metazoneId deprecated (true | false) "false" >
    ***@***.***>
<!ATTLIST metazoneId preferred NMTOKEN #IMPLIED >
    ***@***.***:bcp47/anyvalue-->
    ***@***.***>
<!ATTLIST metazoneId since CDATA "40" >
    ***@***.***:version-->
    ***@***.***>


Mark


On Fri, Sep 3, 2021 at 1:22 PM Yoshito Umaoka ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In common/supplemental/metaZones.xml
> <https://github.com/unicode-org/cldr/pull/1485#discussion_r702143086>:
>
> > @@ -1892,6 +1892,167 @@ For terms of use, see http://www.unicode.org/copyright.html
>  			<mapZone other="Yerevan" territory="001" type="Asia/Yerevan"/>
>  			<mapZone other="Yukon" territory="001" type="America/Whitehorse"/>
>  		</mapTimezones>
> +
> +		<shortIds>
> +            <shortId name="acre" description="Metazone" alias="Acre" since="40"/>
>
> Maybe something like below?
>
> <metaZones>
>   <metazoneList>
>     <metazone type="Acre" short="acre" description="Time for Acre, Brazil"/>
>     ....
>   </metazoneList>
>   <metazoneInfo>
>     <!-- tz/mz mapping data -->
>     ...
>   </metazoneInfo>
> </metaZoens>
>
> If we did not have element, we would want to use <metazones> instead of
> <metazoneList> above. The element describes the definition of each
> metazone - comes with long id and short id.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/unicode-org/cldr/pull/1485#discussion_r702143086>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACJLEMCAFGZXO45K7MXSBVDUAEU7HANCNFSM5DMDU64Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

srl295 commented:
@macchiati Your reply got a little scrambled.

macchiati commented:
BTW, about PathHeader.txt. 

https://cldr.unicode.org/development/updating-dtds has an overview, and we can flesh out as needed.

srl295 commented:
Here's what i'm proposing for the dtd.

- dropped 'description', is there a compelling reason to have a description? We have the localized text in locale data, and we have plenty of metadata about what makes up a metazone. Let me know if there's a use case here.
- changed longId and preferred to `@MATCH:metazone`

```xml
<!ELEMENT metaZones ( metazoneInfo?, mapTimezones?, metazoneIds? ) >

<!ELEMENT metazoneIds ( metazoneId* ) >

<!ELEMENT metazoneId EMPTY >
<!ATTLIST metazoneId shortId NMTOKEN #REQUIRED >
    <!--@MATCH:regex/[a-z][a-z][a-z][a-z]-->
<!ATTLIST metazoneId longId CDATA #IMPLIED >
    <!--@MATCH:metazone-->
    <!--@VALUE-->
<!ATTLIST metazoneId deprecated (true | false) "false" >
    <!--@VALUE-->
<!ATTLIST metazoneId preferred NMTOKEN #IMPLIED >
    <!--@MATCH:metazone-->
    <!--@VALUE-->
<!ATTLIST metazoneId since CDATA "40" >
    <!--@MATCH:version-->
    <!--@METADATA-->
```

srl295 commented:
@macchiati @sffc @yumaoka DTD/XML updated, PTAL

sffc commented on common/supplemental/metaZones.xml:
```diff
@@ -1892,6 +1892,167 @@ For terms of use, see http://www.unicode.org/copyright.html
 		</mapTimezones>
+
+		<metazoneIds>
+            <metazoneId shortId="acre" longId="Acre"/>
```
Nit: Inconsistent space/tab indentation

srl295 replied:
thanks, fixed

sffc commented on common/supplemental/metaZones.xml:
```diff
@@ -1892,6 +1892,167 @@ For terms of use, see http://www.unicode.org/copyright.html
 			<mapZone other="Yukon" territory="001" type="America/Whitehorse"/>
 		</mapTimezones>
+
+		<metazoneIds>
```
Optional: I was thinking originally that this section could contain additional metadata about metazones, not just the ID mapping.  In that case, `<metazoneInfo>` or `<metazoneDetails>` might be an appropriate tag name.

sffc replied:
Thoughts on my optional comment?

```xml
<metazoneDetails>
    <metazoneDetail ... />
    <metazoneDetail ... />
</metazoneDetails>
```

macchiati commented:
We have used the convention on XXXs/XXX in a number of places.

If we had to do it all over again, I would have liked to have changed the
convention to drop the intermediate levels. But given what we have, let's
leave as Steven has it.




Mark


On Mon, Sep 6, 2021 at 4:52 PM Shane F. Carr ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In common/supplemental/metaZones.xml
> <https://github.com/unicode-org/cldr/pull/1485#discussion_r703097369>:
>
> > @@ -1892,6 +1892,167 @@ For terms of use, see http://www.unicode.org/copyright.html
>  			<mapZone other="Yerevan" territory="001" type="Asia/Yerevan"/>
>  			<mapZone other="Yukon" territory="001" type="America/Whitehorse"/>
>  		</mapTimezones>
> +
> +		<metazoneIds>
> +            <metazoneId shortId="acre" longId="Acre"/>
>
> Nit: Inconsistent space/tab indentation
> ------------------------------
>
> In common/supplemental/metaZones.xml
> <https://github.com/unicode-org/cldr/pull/1485#discussion_r703097724>:
>
> > @@ -1892,6 +1892,167 @@ For terms of use, see http://www.unicode.org/copyright.html
>  			<mapZone other="Yerevan" territory="001" type="Asia/Yerevan"/>
>  			<mapZone other="Yukon" territory="001" type="America/Whitehorse"/>
>  		</mapTimezones>
> +
> +		<metazoneIds>
>
> Optional: I was thinking originally that this section could contain
> additional metadata about metazones, not just the ID mapping. In that case,
> <metazoneInfo> or <metazoneDetails> might be an appropriate tag name.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/unicode-org/cldr/pull/1485#pullrequestreview-747447692>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACJLEMAGGRNFXASNHOHWRGDUAVH23ANCNFSM5DMDU64Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

macchiati reviewed (changes requested):
Left comment for fixing the PathHeader failures

macchiati commented:
Whoops, grabbed the wrong one:
//supplementalData/metaZones/metazoneIds/metazoneId[@shortId="%A"]/_%E ; Supplemental ; Metazone ; $1 ; $2 ; HIDE

srl295 commented:
@macchiati @sffc PTAL, should pass tests and be ready to go

macchiati commented:
Just reviewed, looks good!

Mark


On Tue, Sep 7, 2021 at 3:40 PM Steven R. Loomis ***@***.***>
wrote:

> @macchiati <https://github.com/macchiati> @sffc <https://github.com/sffc>
> PTAL, should pass tests and be ready to go
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/unicode-org/cldr/pull/1485#issuecomment-914672659>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACJLEMA2ZCZOYYDRCPKWLMDUA2IEPANCNFSM5DMDU64Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

macchiati commented:
I'd responded 8 hours ago about that.

Mark


On Tue, Sep 7, 2021 at 4:08 PM Shane F. Carr ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In common/supplemental/metaZones.xml
> <https://github.com/unicode-org/cldr/pull/1485#discussion_r703910602>:
>
> > @@ -1892,6 +1892,167 @@ For terms of use, see http://www.unicode.org/copyright.html
>  			<mapZone other="Yerevan" territory="001" type="Asia/Yerevan"/>
>  			<mapZone other="Yukon" territory="001" type="America/Whitehorse"/>
>  		</mapTimezones>
> +
> +		<metazoneIds>
>
> Thoughts on my optional comment?
>
> <metazoneDetails>
>     <metazoneDetail ... />
>     <metazoneDetail ... />
> </metazoneDetails>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/unicode-org/cldr/pull/1485#discussion_r703910602>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACJLEMAUXRWHMXOSXQVWDSTUA2LNHANCNFSM5DMDU64Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

sffc commented:
> I'd responded 8 hours ago about that. Mark

I assume you're taking about this reply?

> We have used the convention on XXXs/XXX in a number of places.
>
> If we had to do it all over again, I would have liked to have changed the
> convention to drop the intermediate levels. But given what we have, let's
> leave as Steven has it.

My optional suggestion is to replace "metazoneIds" with "metazoneDetails", in case we want to add more details than just the IDs in the future.  It is consistent with the "XXXs/XXX" naming convention.

srl295 commented:
@sffc  @macchiati  I'd lke to merge and we can iterate.
